vm_supervisor/run.py

```diff
@@ -75,6 +76,9 @@ async def create_vm_execution(vm_hash: ItemHash) -> VmExecution:
         logger.exception(error)
         pool.forget_vm(vm_hash=vm_hash)
         raise HTTPInternalServerError(reason="Error during runtime initialisation")
+    except HostNotFoundError as error:
+        logger.exception(error)
+        pool.forget_vm(vm_hash=vm_hash)
```

Comment:
Why don't you raise an exception here ?

Reply:
Because if you raise the exception, the supervisor crashes and don't forget the VM, but needs to forget it first.

Reply:
And raising the exception after launching `forget_vm` does not work, as for the other exceptions that are handled ?

Reply:
If we raise an exception after, the rest of the instances will not start it's deploying, and there are sometimes that the instance takes a bit more time to start.

Reply:
The fact that the exception blocks other instances from starting is a good point, but it should not be handled here since any other exception will also block that.